tests/dataimportcron_test.go

```diff
@@ -28,90 +29,118 @@ const (
-		trustedRegistryURL = func() string { return fmt.Sprintf(utils.TrustedRegistryURL, f.DockerPrefix) }
-		dataSourceName     = "datasource-test"
-		cronName           = "cron-test"
-		cron               *cdiv1.DataImportCron
-		err                error
-		ns                 string
+		f              = framework.NewFramework(namespacePrefix)
+		dataSourceName = "datasource-test"
+		cronName       = "cron-test"
+		dvName         = "dv-garbage"
+		cron           *cdiv1.DataImportCron
+		ns             string
 	)
 
 	BeforeEach(func() {
 		ns = f.Namespace.Name
 	})
 
-	table.DescribeTable("should", func(schedule string, retentionPolicy cdiv1.DataImportCronRetentionPolicy, repeat int, checkGarbageCollection bool) {
-		var url string
+	table.DescribeTable("should", func(garbageCollection, retention, createErrorDv bool, repeat int) {
+		reg, err := getDataVolumeSourceRegistry(f)
+		Expect(err).To(BeNil())
+		defer utils.RemoveInsecureRegistry(f.CrClient, *reg.URL)
 
-		if repeat > 1 || utils.IsOpenshift(f.K8sClient) {
-			url = externalRegistryURL
-		} else {
-			url = trustedRegistryURL()
-			err = utils.AddInsecureRegistry(f.CrClient, url)
-			Expect(err).To(BeNil())
+		By(fmt.Sprintf("Create labeled DatVolume %s for garbage collection test", dvName))
+		dv := utils.NewDataVolumeWithRegistryImport(dvName, "5Gi", "")
+		dv.Spec.Source.Registry = reg
+		dv.Labels = map[string]string{common.DataImportCronLabel: cronName}
+		_, err = utils.CreateDataVolumeFromDefinition(f.CdiClient, ns, dv)
+		Expect(err).ToNot(HaveOccurred())
 
-			hasInsecReg, err := utils.HasInsecureRegistry(f.CrClient, url)
-			Expect(err).ToNot(HaveOccurred())
-			Expect(hasInsecReg).To(BeTrue())
-			defer utils.RemoveInsecureRegistry(f.CrClient, url)
+		By(fmt.Sprintf("Create new DataImportCron %s, url %s", cronName, *reg.URL))
+		cron = NewDataImportCron(cronName, "5Gi", scheduleEveryMinute, dataSourceName, *reg)
+		if !garbageCollection {
+			garbageCollect := cdiv1.DataImportCronGarbageCollectNever
+			cron.Spec.GarbageCollect = &garbageCollect
+		}
+		if !retention {
+			retentionPolicy := cdiv1.DataImportCronRetainNone
+			cron.Spec.RetentionPolicy = &retentionPolicy
 		}
-
-		cron = NewDataImportCron(cronName, "5Gi", schedule, dataSourceName, cdiv1.DataVolumeSourceRegistry{URL: &url, PullMethod: &registryPullNode}, retentionPolicy)
-		By(fmt.Sprintf("Create new DataImportCron %s", url))
 		cron, err = f.CdiClient.CdiV1beta1().DataImportCrons(ns).Create(context.TODO(), cron, metav1.CreateOptions{})
 		Expect(err).ToNot(HaveOccurred())
 
-		if schedule == scheduleEveryMinute {
-			By("Verify cronjob was created")
+		By("Verify cronjob was created")
+		Eventually(func() bool {
+			_, err := f.K8sClient.BatchV1().CronJobs(f.CdiInstallNs).Get(context.TODO(), controller.GetCronJobName(cron), metav1.GetOptions{})
+			if errors.IsNotFound(err) {
+				return false
+			}
+			Expect(err).ToNot(HaveOccurred())
+			return true
+		}, dataImportCronTimeout, pollingInterval).Should(BeTrue())
```

Comment:
please add a description here, `Should(BeTrue(), "Cron iob should be created")`

Reply:
Why does it help to add it in each `Eventually().Should()`? doesn't the preceding `By()` tell exactly the same?